mm2src/coins/eth/nft/nft_structs.rs

```diff
@@ -0,0 +1,222 @@
+use crate::{TxFeeDetails, WithdrawFee};
+use mm2_number::BigDecimal;
+use rpc::v1::types::Bytes as BytesJson;
+use serde::Deserialize;
+use std::str::FromStr;
+
+#[derive(Debug, Deserialize)]
+pub struct NftListReq {
+    pub(crate) chains: Vec<Chain>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NftMetadataReq {
+    pub(crate) token_address: String,
+    pub(crate) token_id: BigDecimal,
+    pub(crate) chain: Chain,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
```

Comment:
You can also derive `Copy` here so you can avoid using `clone()` for this data-type.

Reply:
Done